GARE/src/main/java/net/hft/algodat/crossovermethods/CrossoverFunction.java

```diff
@@ -0,0 +1,126 @@
+            indexOfIndividual += 2; // make use of probability here
+        } 
+        LOGGER.info("Crossover permutation is finished");        
+    }   
+    
+    //one point crossover operation
+    private void onePointCrossover(Individual mother, Individual father)
+    {
+        // compute crossover point
+        int crossPoint = getRandomIntegerInBetween(1, mother.jobListe.length-1);
+      
+        // create son and daugther
+        Individual daugther = createChildFromParents(mother, father, crossPoint);
+        Individual son = createChildFromParents(father, mother, crossPoint);
+
+        // substitute parents by children
+        for (int i=0; i < mother.jobListe.length;i++)
+        {
+            mother.jobListe[i] = daugther.jobListe[i];
+            father.jobListe[i] = son.jobListe[i];
+        }
+    }
+
+    // creates child from parent1 and parent2 individuals
+    // crosspoint is passed as parameter
+    // child[0 .. crossPoint-1] jobs copied from parent1
+    // child[crossPoint .. end] feasible jobs copied from parent2
+    // if there are not enough feasible jobs in parent2 then child=parent1
+    private Individual createChildFromParents(Individual parent1, Individual parent2, int crossPoint)
+    {
+        Individual child = new Individual();
+        child.reproduce(parent1); // copy parent1 to child
+
+        int indexInParent = 0;
+        int indexInChild = crossPoint;
+
+        // fill child after crosspoint from parent2
+        while (indexInChild < child.jobListe.length)
+        {
+            // search for first job index in parent2 which is not yet defined in child
+           for(; indexInParent < parent2.jobListe.length; indexInParent++)
+           {
+                if (!isJobNumberAlreadyExist(child.jobListe, 0, indexInChild-1, parent2.jobListe[indexInParent]))
+                {
+                    // insert found parent job index into child
+                    child.jobListe[indexInChild++] = parent2.jobListe[indexInParent];
+                    break;
+                }
+           }
+
+           if (indexInParent >= parent2.jobListe.length)
+           {
+               // stop when parent2 job list processed completelly
+               break;
+           }
+        }
+        // if whole child's job list is filled return child
+        if (indexInChild >= child.jobListe.length)
+            return child;
+
+        // not enough feasible jobs in parent2         
+        return parent1;
+    }
+
+    // Checking if a job number is defined in a specified part of list.Range: [startIndex..endIndex]
+    private boolean isJobNumberAlreadyExist(int[] jobListe, int startIndex, int endIndex, int jobNumber)
+    {
+        boolean flag = false;
+        for(int i=startIndex; i<=endIndex; i++)
+        {
+            if(jobListe[i] == jobNumber)
+                return true;  
+        }
+        return false;
+    } 
+    
+    private static int getRandomIntegerInBetween(int min, int max) {
```

Comment:
This function is already done. Please refactor the method from <code>TournamentSelection</code>. Maybe you can create an <code>abstract</code> Utilities- class